docs/src/doe.md

```diff
@@ -0,0 +1,159 @@
+# DOE Stack
+The Caliptra subsystem supports SPDM, Secure-SPDM over PCI Data Object Exchange (DOE) mailbox protocol. The following diagram gives the over view of the DOE send and receive stack.
+
+DOE Receive stack:
+
+![The DOE Tock receive stack](images/doe_tock_receive.svg)
+
+
+DOE Send stack:
+
+![The DOE Tock send stack](images/doe_tock_send.svg)
+
+```mermaid
+sequenceDiagram
+    participant Host as "Host(TSM)"
+    participant SoC_PCI_DOE_FSM as "SoC PCI DOE Listener"
+    participant MCU_DOE_MBOX_DRIVER as "MCU DOE Mailbox Driver"
+    participant DOE_CAPSULE as "DOE Capsule"
+    participant SPDM_APP as "SPDM App"
+    SPDM_APP -->> DOE_CAPSULE: App invokes receive_message<br> for SPDM or Secure-SPDM Data Object type 
+    Host ->> Host: Host waits until<br> the `DOE Busy` bit is cleared in <br>DOE Status Register
+    loop While there is remaining DOE object to send
+        Host ->> SoC_PCI_DOE_FSM  : Host starts sending DOE data object
+        SoC_PCI_DOE_FSM ->> SoC_PCI_DOE_FSM: Prepare the message <br>in DOE Mailbox SRAM
+        Note right of Host: Repeat until Host sets DOE Go bit
+        Host ->> SoC_PCI_DOE_FSM: Host writes `DOE Go` bit <br>in DOE Control Register<br> to indicate message ready
+    end
+    SoC_PCI_DOE_FSM ->> MCU_DOE_MBOX_DRIVER: Notify that a new DOE object<br> is available to consume
+    MCU_DOE_MBOX_DRIVER ->> DOE_CAPSULE: receive() callback
+
+    alt if DOE object is `Data Object 0`
+        DOE_CAPSULE ->> DOE_CAPSULE: Copy DOE object payload <br>into local buffer
+        DOE_CAPSULE ->> DOE_CAPSULE: Handle DOE Discovery
```

Comment:
Instead of having the capsule handle this logic, would it make sense to have the application handle it?

That way, the DOE capsule can just be a generic mailbox capsule.